### DISCUSSION TITLE
GitHub Apps Do Not Have Permissions to Execute the GraphQL UpdateRepository Mutation

### DISCUSSION THREAD

**Discussion by Syntex9000** [community/community#67592] [Apps, API and Webhooks] [answered]
### Select Topic Area

Bug

### Body

I have a GitHub app that authenticates as an installation with full-blown access to the repository and organization permissions. When I try to execute the GraphQL updateRepository mutation to set the hasDiscussionsEnabled flag (or any repository input, for that matter), I am greeted with the following result (200 OK)

Request:

```
mutation UpdateRepository {
  updateRepository(input: 
    { 
      repositoryId:"redacted", hasDiscussionsEnabled:true 
    }) 
  {
    repository {
      hasDiscussionsEnabled
      name
    }
    clientMutationId
  }   
}
```

Result:
```
{
    "data": {
        "updateRepository": null
    },
    "errors": [
        {
            "type": "UNPROCESSABLE",
            "path": [
                "updateRepository"
            ],
            "locations": [
                {
                    "line": 3,
                    "column": 3
                }
            ],
            "message": "my-service does not have permission to toggle discussions on {org/repo}."
        }
    ]
}
```

However, I can authenticate with my user account and execute the query without issue. Since there is no equivalent REST API call to enable discussions, this effectively denies GitHub Apps any way of toggling discussions on or off programmatically. 

Perhaps I am missing something?

**Comment by gm3dmo** [comment, marked as answer]
> Since there is no equivalent REST API call to enable discussions, this effectively denies GitHub Apps any way of toggling discussions on or off programmatically.

The [update-a-repository](https://docs.github.com/en/rest/reference/repos#update-a-repository) REST API endpoint support `has_discussions`

```
{
  "has_discussions": true
}
```

```
 curl --no-progress-meter -X PATCH -H 'Accept: application/vnd.github.v3+json' -H 'Authorization: Bearer ghs_***' https://api.github.com/repos/forest-town/repo-discussions-67592 --data @tmp/update-a-repo.json
```

Can you try something like that?

**Reply by Syntex9000** [reply]
Son of a gun, the documentation for the REST API is out of date. Thank you!

**Reply by gm3dmo** [reply]
@dirtygooback 

> Son of a gun, the documentation for the REST API is out of date. Thank you!

It's often worth having a peep at the current output of [get-a-repository](https://docs.github.com/en/rest/reference/repos#get-a-repository) because the samples in the documentation are not "live".

Here's an example:

```
 ./list-repo.sh | gron | grep has_
```

``
json.has_discussions = false;
json.has_downloads = true;
json.has_issues = true;
json.has_pages = false;
json.has_projects = true;
json.has_wiki = true;
```

[create-repo-testrepo.sh](https://github.com/gm3dmo/the-power/blob/main/create-repo-testrepo.sh)

I try to maintain and keep current a set of samples in [the-power](https://github.com/gm3dmo/the-power/) where [create-repo-testrepo.sh](https://github.com/gm3dmo/the-power/blob/main/create-repo-testrepo.sh) will create a repository which has a chosen set of features.

**Reply by Syntex9000** [reply]
I am starting to see that I can't trust the docs to be current, which is a shame.

**Comment by Syntex9000** [comment]
Although this fixed my issue, the GraphQL problem above is still a thing and I have a ticket into GitHub Support. They have acknowledged it is a bug.

**Comment by UmeshMehta1** [comment]
great
